homeassistant/components/home_connect/api.py

```diff
@@ -211,14 +213,34 @@ def get_opstate_sensor(self):
 class DeviceWithDoor(HomeConnectDevice):
     """Device that has a door sensor."""
 
-    def get_door_entity(self):
+    def get_door_entities(self):
         """Get a dictionary with info about the door binary sensor."""
-        return {
-            ATTR_DEVICE: self,
-            ATTR_DESC: "Door",
-            ATTR_SENSOR_TYPE: "door",
-            ATTR_DEVICE_CLASS: "door",
-        }
+        entities = []
+        status_keys = self.appliance.get_status().keys()
+        for key in status_keys:
+            if key == BSH_DOOR_STATE:
+                entities.append(
+                    {
+                        ATTR_DEVICE: self,
+                        ATTR_DESC: "Door",
+                        ATTR_KEY: key,
+                        ATTR_SENSOR_TYPE: "door",
+                        ATTR_DEVICE_CLASS: "door",
+                    }
+                )
+            elif REFRIGATION_DOOR_KEY_PREFIX in key:
+                device_name = key.replace(REFRIGATION_DOOR_KEY_PREFIX, "")
+                desc = f"{device_name} Door"
+                entities.append(
+                    {
+                        ATTR_DEVICE: self,
+                        ATTR_DESC: desc,
```

Comment:
similarly, 
desc is only used once - just access directly?
`ATTR_DESC: f"{device_name} Door"`

or go "all the way" eliminating device_name with:
`ATTR_DESC: f'{key.replace(REFRIGATION_DOOR_KEY_PREFIX, "")} Door'`